python/GafferSceneUI/SceneViewUI.py

```diff
@@ -1073,13 +1076,105 @@ def __appendClippingPlaneMenuItems( menuDefinition, prefix, view, parentWidget )
 			}
 		)
 
+def __snapshotDescription( view ) :
+
+	sceneGadget = view.viewportGadget().getPrimaryChild()
+	if sceneGadget.getRenderer() == "OpenGL" :
+		return "Viewport snapshots are only available for rendered (non-OpenGL) previews."
+
+	return "Snapshot viewport and send to catalogue."
+
+def __snapshotToCatalogue( catalogue, view ) :
+
+	timeStamp = str( datetime.datetime.now() )
+
+	fileName = pathlib.Path(
+		view["in"].node().getContext().substitute( catalogue["directory"].getValue() )
+	) / ( timeStamp.replace( ":", "-" ).replace(" ", "_" ) + ".exr" )
```

Comment:
Should we use a `viewerSnapshot` prefix or something like that in the filename, to give a bit more of a clue about where the file came from?

Reply:
Added in https://github.com/GafferHQ/gaffer/pull/5533/commits/792babe29290021ed8936aafab31039f06386bfe